lib/github.ml

```diff
@@ -14,14 +14,21 @@ module D = struct
 end
 
 module Parse = struct
+  let user_from_regexp_opt uri regexp =
+    try Some (Re.(Group.get (exec regexp uri) 1))
+    with Not_found -> None
+
   let user_from_remote remote_uri =
     let ssh_uri_regexp =
       Re.Emacs.compile_pat "git@github\\.com:\\(.+\\)/.+\\(\\.git\\)?"
     in
-    try
-      let substrings = Re.exec ssh_uri_regexp remote_uri in
-      Some (Re.Group.get substrings 1)
-    with Not_found -> None
+    match user_from_regexp_opt remote_uri ssh_uri_regexp with
```

Comment:
How about disambiguating the `https` vs `ssh` thing earlier rather than trying to parse as an ssh URL and fallback to the https case?

That will also make it easier to add different cases in the future if we ever need to!

Reply:
done